src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/CollectionExerciseServiceImpl.java

```diff
@@ -217,17 +222,91 @@ private void setCollectionExerciseFromDto(CollectionExerciseDTO collex, Collecti
     public CollectionExercise createCollectionExercise(CollectionExerciseDTO collex) {
-        CollectionExercise collectionExercise = new CollectionExercise();
+        log.debug("Attempting to create collection exercise with action plans");
+        CollectionExercise collectionExercise = newCollectionExerciseFromDTO(collex);
+        // Save collection exercise before creating action plans because we need exercisepk
+        collectionExercise = this.collectRepo.saveAndFlush(collectionExercise);
+        createActionPlans(collectionExercise);
+
+        log.debug("Successfully created collection exercise and action plans.");
+        return collectionExercise;
+    }
 
+    /**
+     * Create and populate details of collection exercise.
+     * @param collex collection exercise
+     * @return collection exercise with details
+     */
+    private CollectionExercise newCollectionExerciseFromDTO(CollectionExerciseDTO collex) {
+        log.debug("Create new collection exercise from DTO");
+        CollectionExercise collectionExercise = new CollectionExercise();
         setCollectionExerciseFromDto(collex, collectionExercise);
-
         collectionExercise.setState(CollectionExerciseDTO.CollectionExerciseState.CREATED);
         collectionExercise.setCreated(new Timestamp(new Date().getTime()));
         collectionExercise.setId(UUID.randomUUID());
 
-        return this.collectRepo.saveAndFlush(collectionExercise);
+        log.debug("Successfully created collection exercise from DTO");
+        return collectionExercise;
+    }
+
+    /**
+     * Set up data for creation of action plans
+     * @param collectionExercise Collection Exercise
+     */
+    private void createActionPlans(CollectionExercise collectionExercise) {
+        log.debug("Attempting to create action plans for Collection exercise %s", collectionExercise.getId());
+        SurveyDTO survey = surveyService.findSurvey(collectionExercise.getSurveyId());
+        String shortName = survey.getShortName();
+
+        createActionPlan(shortName, collectionExercise, "B");
+        createActionPlan(shortName, collectionExercise, "BI");
+        log.debug("Successfully created action plans for Collection Exercise %s", collectionExercise.getId());
+    }
+
+    /**
+     * Create action plan for collection exercise and case type
+     * @param shortName Survey Short Name
+     * @param collectionExercise Collection Exercise
+     * @param caseType Sample Unit Type
+     */
+    private void createActionPlan(String shortName, CollectionExercise collectionExercise, String caseType) {
+        log.debug("Attempting to create action plan and case type overrides for collection exercise %s case type, %s",
+                   collectionExercise.getId(), caseType);
+        String exerciseRef = collectionExercise.getExerciseRef();
+        String name = shortName + " " + caseType + " " + exerciseRef;
+        String description = shortName + " " + caseType + " Case " + exerciseRef;
```

Comment:
Alternatively but not necessarily better... String.format("%s %s Case %s", shortName, caseType, exerciseRef)